### PR TITLE
Resp clamper updates

### PR DESCRIPTION
**Pull request by rdallman**
* moves clamper to only on the stdout writer (from protocol)
* adds buffering for all formats, so that we can do ^ for cold and removes the shared protocol buffer
* fixes bug where docker i/o errors weren't plumbed up from container wait
* changes clamp writer to allow specifying an error to return so we don't need to check every call site for a type cast and a bool bit, we were throwing off an io.EOF anyway so tiny behavior change here [placement aside].
* shuts down a hot container when output goes over limit, where previously we would non-deterministically try to flush the pipe for a small period of time. attempts at fixing this were futile, it wound up in adding an extra buffered io writer on top just so that we could check for terminal bits in the encodings, just not gonna go that far -- lying about bytes written also didn't work.

### CODE REVIEW

File: docs/swagger.yml [hunk 1]
```diff
@@ -553,13 +553,6 @@ definitions:
...
         readOnly: true
 
-  Version:
-    type: object
-    properties:
-      version:
-        type: string
-        readOnly: true
-
   RoutesWrapper:
```
Comment (skinowski) [inline docs/swagger.yml:556]: Hmm, why did this get removed?

**Comment by skinowski** [comment]
Couple things in there, version, default size, etc, but I'll clean up...

File: docs/swagger.yml (hunk 1 above)
Comment (rdallman) [reply docs/swagger.yml:556]: this was from the merge commit. rebase attempts were not fun, sorry

**Comment by rdallman** [comment]
thanks. imo by default this should be off. when someone wants to run the communist version and impose all the rules they'll just have to turn stuff like this on

**Comment by skinowski** [comment]
I agree, I meant fixing up config.go, which has 10 MB.

**Pull request merged**